checkov/common/checks_infra/solvers/attribute_solvers/base_attribute_solver.py

```diff
@@ -151,10 +151,13 @@ def _process_node(
         else:
             failed_vertices.append(data)
 
+    def is_needs_all_condition(self) -> bool:
```

Comment:
can you add documentation to when should we override this?
also - I think grammar wise it is better to use `should_check_all_condition`